Executables/Game/World/Entities/PlayerEntity.cs

```diff
@@ -140,10 +147,54 @@ public class PlayerEntity : Entity, IPlayerEntity
             Inventory = new Inventory(itemManager, db, _cacheManager, _logger, player.Id, 1, 5, 9, 2);
             QuickSlotBar = new QuickSlotBar(_cacheManager, _logger, this);
 
-            MovementSpeed = 150;
+            MovementSpeed = GetMovementSpeed();
             EntityClass = player.PlayerClass;
 
             Groups = new List<Guid>();
+            Affects = new List<Affect>();
+        }
+
+        public byte GetMovementSpeed()
+        {
+            var totalApplyValue = 0;
+            if (Affects != null && Affects.Count > 0)
+            {
+                totalApplyValue = Affects.Where(a => a.ApplyOn == (int)EPoints.MoveSpeed).Sum(a => a.ApplyValue);
```

Comment:
Avoid using LINQ here. This might be a hot path. We want to avoid as many allocations as possible